### PR TITLE
Wait to Activate Editor-server Until the Logging System is Ready

### PR DESCRIPTION
**Pull request by AMZN-Gene**
Wait to activate the editor-server until CrySystemInitialized so that the logging system is ready. Automated testing listens for these logs

### CODE REVIEW

File: Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp [hunk 1]
```diff
@@ -34,9 +35,24 @@ namespace Multiplayer
         m_networkEditorInterface = AZ::Interface<INetworking>::Get()->CreateNetworkInterface(
             AZ::Name(MpEditorInterfaceName), ProtocolType::Tcp, TrustZone::ExternalClientToServer, *this);
         m_networkEditorInterface->SetTimeoutMs(AZ::TimeMs{ 0 }); // Disable timeouts on this network interface
-        ActivateDedicatedEditorServer();
+        CrySystemEventBus::Handler::BusConnect();
```
Comment (AMZN-puvvadar) [inline Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp:38]: Any chance CrySystem is already initialized at this point?

Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp:38]: Good question! 
For Game/Server Launchers the system components get turned on during Launcher.cpp Run() method where is says gameApplication.Start(). . . below that logic is where the CrySystem is created.

For Editor the magic happens in CryEditMain: here the system components are created deep inside AZToolsApp.Start() and below that there's the call to crysystem init deep inside CCryEditApp::instance()->InitInstance

In a perfect world, the logging system would have instead been attached to a system component which my multiplayer system component would have easily been able to require service.

Comment (jackalbe) [reply Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp:38]: @AMZN-stankowi did some work with the life cycle of the application recently that signals the application events for handlers to listen for

File: Gems/Multiplayer/Code/CMakeLists.txt [hunk 2]
```diff
@@ -25,6 +25,7 @@ ly_add_target(
             AZ::AzCore
             AZ::AzFramework
             AZ::AzNetworking
+            Legacy::CryCommon
```
Comment (jackalbe) [inline Gems/Multiplayer/Code/CMakeLists.txt:28]: we really do not want to add dependencies to CryCommon, maybe use a more recent system?

Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/CMakeLists.txt:28]: Multiplayer gem was already relying on CryCommon, I just moved in the cmake so I could use it for this purpose :c

File: Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp (hunk 1 above)
Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp:38]: Nice! I'll follow up with @AMZN-stankowi

Comment (AMZN-puvvadar) [reply Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp:38]: Second this, if we can avoid relying on CrySystemEventBus that'd be great

File: Gems/Multiplayer/Code/CMakeLists.txt (hunk 2 above)
Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/CMakeLists.txt]: Fixed! Updated to use the new ComponentApplicationLifecycle

File: Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp (hunk 1 above)
Comment (AMZN-Gene) [reply Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp]: Fixed! Using the new ComponentApplicationLifecycle

File: Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp [hunk 3]
```diff
@@ -34,9 +36,27 @@ namespace Multiplayer
...
         m_networkEditorInterface->SetTimeoutMs(AZ::TimeMs{ 0 }); // Disable timeouts on this network interface
-        ActivateDedicatedEditorServer();
-    }
 
+        // Wait to activate the editor-server until LegacySystemInterfaceCreated so that the logging system is ready
+        // Automated testing listens for these logs
+        if (editorsv_isDedicated)
+        {
+            // If the settings registry is not available at this point,
+            // then something catastrophic has happened in the application startup.
```
Comment (AMZN-stankowi) [inline Gems/Multiplayer/Code/Source/Editor/MultiplayerEditorConnection.cpp:45]: I regret writing this comment now. It's going to show up everywhere as people wire in to the application lifecycle events.

No change necessary, I'm just taking a note for myself in the future to expect this to happen more often.

**Review by AMZN-stankowi — APPROVED** [review]

**Review by AMZN-puvvadar — APPROVED** [review]

**Pull request merged**